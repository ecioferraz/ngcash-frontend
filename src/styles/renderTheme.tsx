import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (children: ReactNode): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
