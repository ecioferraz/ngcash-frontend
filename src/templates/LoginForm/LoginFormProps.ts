// import { ReactNode } from 'react';

// type LoginFormProps = { children: ReactNode };

// export default LoginFormProps;
