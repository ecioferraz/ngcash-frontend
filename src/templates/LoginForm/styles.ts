import styled, { css } from 'styled-components';

export const Container = styled.form`
  ${({ theme }) => css`
    align-items: center;
    backdrop-filter: blur(2px);
    background-color: rgba(255, 255, 255, 0.05);
    border-radius: 15px;
    box-shadow: rgba(0, 0, 0, 0.2) 0px 12px 28px 0px, rgba(0, 0, 0, 0.1) 0px 2px 4px 0px, rgba(255, 255, 255, 0.05) 0px 0px 0px 1px inset;
    display: flex;
    flex-direction: column;
    justify-content: center;
    -webkit-backdrop-filter: blur(6px);
    padding: 1em 2em;
    width: fit-content;

    > h1 {
      text-align: center;
      width: 100%;
    }

    > * {
      display: flex;
      flex-direction: column;
      margin: 1rem;
    }

    input {
      background: transparent;
      border: none;
      border-bottom: solid 1px ${theme.colors.mediumGray};
      color: ${theme.colors.white};
      font-size: ${theme.font.sizes.small};
      outline: none;
    }

    input:focus {
      border: none;
      border-bottom: 1px solid ${theme.colors.lightGreen};
      outline: none;
    }

    section > * {
      margin: 0.3rem
    }

    button {
      background: ${theme.colors.black};
      border: 1px solid ${theme.colors.black};
      border-radius: 4px;
      box-shadow: ${theme.colors.mediumGray} 3px 3px 0 0,${theme.colors.black} 3px 3px 0 1px;
      box-sizing: border-box;
      color: ${theme.colors.mediumGray};
      cursor: pointer;
      display: inline-block;
      line-height: 20px;
      overflow: visible;
      padding: 6px 24px;
      touch-action: manipulation;
      user-select: none;
      -webkit-user-select: none;
      vertical-align: middle;
      white-space: nowrap;
    }

    button:disabled {
      pointer-events: none;
    }

    button:focus {
      text-decoration: none;
    }

    button:hover {
      text-decoration: none;
    }

    button:active {
      background: ${theme.colors.lightGreen};
      box-shadow: rgba(0, 0, 0, 0.125) 0 3px 5px inset;
      outline: 0;
      transition: none;
    }

    button:not([disabled]):active {
      box-shadow: ${theme.colors.mediumGray} 2px 2px 0 0, ${theme.colors.black} 2px 2px 0 1px;
      translate: 2px, 2px;

    }

    @media (min-width: 768px) {
      button {
        padding: 12px 50px;
      }
    }
  `}
`;
