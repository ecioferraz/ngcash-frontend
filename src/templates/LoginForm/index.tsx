import Button from '../../components/Button';
import Heading from '../../components/Heading';
import TextInput from '../../components/TextInput';
import * as Styled from './styles';

export default function LoginForm() {
  return (
    <Styled.Container>
      <Heading>Bem-vindo(a)!</Heading>
      <TextInput
        handleChange={() => {
          return;
        }}
        placeholder="Username"
      />
      <TextInput
        handleChange={() => {
          return;
        }}
        placeholder="Password"
      />
      <Button
        handleClick={() => {
          return;
        }}
      >
        Login
      </Button>
      <Button
        handleClick={() => {
          return;
        }}
      >
        Register
      </Button>
    </Styled.Container>
  );
}
