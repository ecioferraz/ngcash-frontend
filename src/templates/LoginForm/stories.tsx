import { Meta, Story } from '@storybook/react/types-6-0';
// import LoginFormProps from './LoginFormProps';
import LoginForm from '.';

export default {
  component: LoginForm,
  title: 'Templates/LoginForm',
} as Meta;

export const Template: Story = (args) => <LoginForm {...args} />;
