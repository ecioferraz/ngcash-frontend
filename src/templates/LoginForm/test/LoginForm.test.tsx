import renderTheme from '../../../styles/renderTheme';
import LoginForm from '..';

describe('<LoginForm />', () => {
  it('should render <LoginForm />', () => {
    const { container } = renderTheme(<LoginForm />);

    expect(container).toMatchSnapshot();
  });
});
