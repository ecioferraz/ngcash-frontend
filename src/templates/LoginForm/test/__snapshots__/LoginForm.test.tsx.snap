// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render <LoginForm /> 1`] = `
.c1 {
  background: #dc143c;
}

.c0 {
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<div>
  <form
    class=""
  >
    <h1
      class="c0"
    >
      Bem-vindo(a)!
    </h1>
    <input
      class=""
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      class=""
      placeholder="Password"
      type="text"
      value=""
    />
    <button
      class="c1"
      type="button"
    >
      Login
    </button>
    <button
      class="c1"
      type="button"
    >
      Register
    </button>
  </form>
</div>
`;
